test/unit/math/mix/fun/generalized_inverse_test.cpp

```diff
@@ -40,10 +76,24 @@ TEST(mathMixMatFun, ad_tests) {
   w << 2, 3, 5, 7, 11, 13, 17, 19, 23, 25, 27, 29;
   expect_ad(tols, f, w);
   expect_ad_matvar(f, w);
+}
+
+TEST(mathMixMatFun, ad_tests8) {
+  using stan::test::expect_ad;
+  using stan::test::expect_ad_matvar;
+
+  auto f = [](const auto& G) { return stan::math::generalized_inverse(G); };
 
   Eigen::MatrixXd z(2, 2);
   z << 1, 2, 5, std::numeric_limits<double>::quiet_NaN();
   EXPECT_NO_THROW(stan::math::generalized_inverse(z));
+}
+
+TEST(mathMixMatFun, ad_tests9) {
```

Comment:
last quick thing, just change these to have like the second name like gen_inv_test_{#} instead of `ad_tests{#}`. Then I think this is good!